x/autopilot/keeper/redeem_stake_test.go

```diff
@@ -0,0 +1,253 @@
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // invalid redeem receiver
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "xxx", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+	}
+
+	for i, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %d", i), func() {
+			packet.Data = transfertypes.ModuleCdc.MustMarshalJSON(&tc.packetData)
+
+			suite.SetupTest() // reset
+			ctx := suite.Ctx
+
+			suite.App.AutopilotKeeper.SetParams(ctx, types.Params{StakeibcActive: tc.forwardingActive})
+
+			// set epoch tracker for env
+			suite.App.StakeibcKeeper.SetEpochTracker(ctx, stakeibctypes.EpochTracker{
+				EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+				EpochNumber:        1,
+				NextEpochStartTime: uint64(now.Unix()),
+				Duration:           43200,
+			})
+			suite.App.StakeibcKeeper.SetEpochTracker(ctx, stakeibctypes.EpochTracker{
+				EpochIdentifier:    "day",
+				EpochNumber:        1,
+				NextEpochStartTime: uint64(now.Unix()),
+				Duration:           86400,
+			})
+			// set deposit record for env
+			suite.App.RecordsKeeper.SetDepositRecord(ctx, recordstypes.DepositRecord{
+				Id:                 1,
+				Amount:             sdk.NewInt(100),
+				Denom:              atomIbcDenom,
+				HostZoneId:         "hub-1",
+				Status:             recordstypes.DepositRecord_TRANSFER_QUEUE,
+				DepositEpochNumber: 1,
+				Source:             recordstypes.DepositRecord_STRIDE,
+			})
+
+			suite.App.RecordsKeeper.SetEpochUnbondingRecord(ctx, recordstypes.EpochUnbondingRecord{
+				EpochNumber: 1,
+				HostZoneUnbondings: []*recordstypes.HostZoneUnbonding{
+					{
+						HostZoneId:            "hub-1",
+						Status:                recordstypes.HostZoneUnbonding_CLAIMABLE,
+						UserRedemptionRecords: []string{},
+						NativeTokenAmount:     sdk.NewInt(1000000),
+					},
+				},
+			})
+
+			// set host zone for env
+			suite.App.StakeibcKeeper.SetHostZone(ctx, stakeibctypes.HostZone{
+				ChainId:               "hub-1",
+				ConnectionId:          "connection-0",
+				Bech32Prefix:          "cosmos",
+				TransferChannelId:     "channel-0",
+				Validators:            []*stakeibctypes.Validator{},
+				BlacklistedValidators: []*stakeibctypes.Validator{},
+				WithdrawalAccount:     nil,
+				FeeAccount:            nil,
+				DelegationAccount:     nil,
+				RedemptionAccount:     nil,
+				IbcDenom:              atomIbcDenom,
+				HostDenom:             atomHostDenom,
+				RedemptionRate:        sdk.NewDec(1),
+				Address:               addr1.String(),
+				StakedBal:             sdk.NewInt(1000000),
+			})
```

Comment:
I think most of this state is not required for this test. I'm happy to take a pass at slimming it down if you want!

Reply:
resolved in #1030